Library/Homebrew/cmd/list.rb

```diff
@@ -20,26 +20,26 @@ def list_args
         List all installed formulae and casks.
 
         If <formula> is provided, summarise the paths within its current keg.
+        If <cask> is provided, list its artifacts.
       EOS
       switch "--formula", "--formulae",
-             description: "List only formulae."
+             description: "List only formulae, or treat all named arguments as formulae."
       switch "--cask", "--casks",
-             description: "List only casks, or <cask> if provided."
+             description: "List only casks, or treat all named arguments as casks."
       switch "--unbrewed",
              description: "List files in Homebrew's prefix not installed by Homebrew."
```

Comment:
```suggestion
      switch "--unbrewed",
             description: "List files in Homebrew's prefix not installed by Homebrew."
```

I wonder if a `deprecated` flag or similar could:
- set `odeprecated` if this flag is called
- stop it from being shown in the manpage and help.

Reply:
I added a `replacement:` option to `switch()` to pass a suggestion to `odisabled` if the switch is used, but I wasn't able to hide it from the `--help` output since that comes directly from `OptionParser`.